cluster-autoscaler/cloudprovider/oci/oke/oci/oci_cloud_provider.go

```diff
@@ -0,0 +1,157 @@
+	if errors.Cause(err) == ErrInstanceNodePoolNotFound {
+		return nil, nil
+	}
+	return ng, err
+}
+
+// GetNodeGpuConfig returns the label, type and resource name for the GPU added to node. If node doesn't have
+// any GPUs, it returns nil.
+func (ocp *OciCloudProvider) GetNodeGpuConfig(node *apiv1.Node) *cloudprovider.GpuConfig {
+	return gpu.GetNodeGPUFromCloudProvider(ocp, node)
+}
+
+// HasInstance returns whether a given node has a corresponding instance in this cloud provider
+func (ocp *OciCloudProvider) HasInstance(n *apiv1.Node) (bool, error) {
+	return true, cloudprovider.ErrNotImplemented
+}
+
+// Pricing returns pricing model for this cloud provider or error if not available.
+// Implementation optional.
+func (ocp *OciCloudProvider) Pricing() (cloudprovider.PricingModel, caerrors.AutoscalerError) {
+	klog.Info("Pricing called")
+	return nil, cloudprovider.ErrNotImplemented
+}
+
+// GetAvailableMachineTypes get all machine types that can be requested from the cloud provider.
+// Implementation optional.
+func (ocp *OciCloudProvider) GetAvailableMachineTypes() ([]string, error) {
+	klog.Info("GetAvailableMachineTypes called")
+	return nil, cloudprovider.ErrNotImplemented
+}
+
+// NewNodeGroup builds a theoretical node group based on the node definition provided. The node group is not automatically
+// created on the cloud provider side. The node group is not returned by NodeGroups() until it is created.
+// Implementation optional.
+func (ocp *OciCloudProvider) NewNodeGroup(machineType string,
+	labels map[string]string,
+	systemLabels map[string]string,
+	taints []apiv1.Taint,
+	extraResources map[string]resource.Quantity,
+) (cloudprovider.NodeGroup, error) {
+	return nil, cloudprovider.ErrNotImplemented
+}
+
+// GetResourceLimiter returns struct containing limits (max, min) for resources (cores, memory etc.).
+func (ocp *OciCloudProvider) GetResourceLimiter() (*cloudprovider.ResourceLimiter, error) {
+	return ocp.rl, nil
+}
+
+// GPULabel returns the label added to nodes with GPU resource.
+func (ocp *OciCloudProvider) GPULabel() string {
+	return ""
+}
+
+// GetAvailableGPUTypes return all available GPU types cloud provider supports.
+func (ocp *OciCloudProvider) GetAvailableGPUTypes() map[string]struct{} {
+	return map[string]struct{}{}
+}
+
+// Cleanup cleans up open resources before the cloud provider is destroyed, i.e. go routines etc.
+func (ocp *OciCloudProvider) Cleanup() error {
+	return ocp.manager.Cleanup()
+}
+
+// Refresh is called before every main loop and can be used to dynamically update cloud provider state.
+// In particular the list of node groups returned by NodeGroups can change as a result of CloudProvider.Refresh().
+func (ocp *OciCloudProvider) Refresh() error {
+	return ocp.manager.Refresh()
+}
+
+func BuildOci(opts config.AutoscalingOptions, do cloudprovider.NodeGroupDiscoveryOptions, rl *cloudprovider.ResourceLimiter) *OciCloudProvider {
+	manager, err := CreateOciManager(nil, do, createKubeClient(opts))
```

Comment:
Configuration parameters that make sense for both implementations should be honored. Our [README](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/oci/README.md#optional-cloud-config-file) documents the parameters that can be set via `--cloud-config`. However, the OKE implementation is dropping the config here: `CreateOciManager(nil, do, createKubeClient(opts))`.

Reply:
completed with current refactor